include/skipper/sequential_map.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef SKIPPER_SEQUENTIAL_MAP_HPP
+#define SKIPPER_SEQUENTIAL_MAP_HPP
+
+#include <memory>
+#include <vector>
+#include <tuple>
+
+namespace skipper {
+
+template <typename Key, typename Value>
+class SequentialSkipListMap {
+ private:
+  struct Node;
+
+ public:
+  using Level = int;
+  using Probability = double;
+
+  static constexpr auto kMaxLevel = Level{4};
+  static constexpr auto kProbability = Probability{0.2};
+
+ public:
+  class Element {
+   public:
+    Key key;
+    Value value;
+
+   public:
+    Element(Key&& k, Value&& v) : key(std::move(k)), value(std::move(v)) {
+    }
+  };
+
+  class Iterator {
+   public:
+    explicit Iterator(Node* ptr);
+
+    auto operator*() -> Element&;
+    auto operator*() const -> const Element&;
+    auto operator->() -> Element*;
+    auto operator->() const -> const Element*;
+    auto operator++(/* prefix */) -> Iterator&;
+    auto operator++(int /* postfix */) -> Iterator;
+    auto operator==(const Iterator& other) const -> bool;
+    auto operator!=(const Iterator& other) const -> bool;
+
+   private:
+    Node* ptr_;
+  };
+
+ public:
+  SequentialSkipListMap() = default;
```

Comment:
Add destructor similar to `SequentialSkipListSet` and delete copy/move constructors/`operator=`.